precompiles/preimage/src/tests.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+use crate::mock::*;
+use precompile_utils::testing::*;
+
+#[test]
+fn test_solidity_interface_has_all_function_selectors_documented_and_implemented() {
+	for file in ["Preimage.sol"] {
+		for solidity_fn in solidity::get_selectors(file) {
```

Comment:
Whenever I see for loops in unit tests, my first thought is always about what the test would do if the loops stopped running as intended.

For example, if `Preimage.sol` were renamed, would this test silently stop doing its job (succeeding even though it started running 0 iterations)?

Reply:
> if renamed, would test silently fail

Not sure but I used this test and all tests like this to get the Solidity selectors so I know it's not failing now. This sounds like a reasonable follow up for all precompiles with this test...